File: librarymanager-be/src/main/java/ru/elenacherev/librarymanager/mapper/GenreMapper.java
package ru.elenacherev.librarymanager.mapper;

import ru.elenacherev.librarymanager.api.dto.Genre;
import ru.elenacherev.librarymanager.domain.entity.GenreEntity;

public class GenreMapper {
    public static Genre map(GenreEntity entity) {
        return Genre.builder()
                .genreId(entity.getGenreId())
                .title(entity.getTitle())
                .build();
    }

    public static GenreEntity map(Genre dto) {
        return map(dto, new GenreEntity());
    }

    public static GenreEntity map(Genre dto, GenreEntity entity) {
        entity.setTitle(dto.getTitle());
        return entity;
    }
}

File: librarymanager-be/src/main/kotlin/ru/elenacherev/librarymanager/mapper/GenreMapper.kt
package ru.elenacherev.librarymanager.mapper

import ru.elenacherev.librarymanager.api.dto.Genre
import ru.elenacherev.librarymanager.domain.entity.GenreEntity

fun GenreEntity.map() = Genre(
    genreId = genreId,
    title = title
)
fun Genre.map() = GenreEntity(title = title)

fun Genre.map(entity:GenreEntity) = entity.also {
    it.title = title
}
